client/my-sites/controller.js

```diff
@@ -548,11 +548,26 @@ export function siteSelection( context, next ) {
 	const isUnlinkedCheckoutNotBoost =
 		isUnlinkedCheckout && ! context.pathname.includes( 'jetpack_boost' );
 
+	const currentPlanSlug = getSitePlanSlug( getState(), siteId );
+	const shouldUpdateStateAfterUpgrade =
+		context.pathname.includes( 'plans/my-plan/trial-upgraded/' ) &&
```

Comment:
Noting that this could be made more precise. Right now, it would return `true` for `my-plans/my-plan/trial-upgraded/` so maybe we want to change the `.includes()` with a `.startsWith()`. You might need to add a leading forward slash then.